### PR TITLE
[lldb-moduleimport-test] Add a way to test getTypeFromMangledSymbolName.

### PR DESCRIPTION
**Pull request by dcci**
The only way we had to test this path was through swift-ide-test,
which takes an input a source file, instead of a serialized module.

This is not the scenario that lldb tests, hence this patch.

<rdar://problem/38323564>

### CODE REVIEW

**Comment by dcci** [comment]
@swift-ci please test

File: test/DebugInfo/DumpTypeFromMangledName.swift [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+// RUN: %lldb-moduleimport-test %t/ASTSection \
+// RUN:   -type-from-mangled='%t/input.sym' 2>&1 | %FileCheck %s
+
+// REQUIRES: executable_test
+func main() {
+  struct Patatino {}
+}
+
+// CHECK: Type of $S4blah4mainyyF
```
Comment (adrian-prantl) [inline test/DebugInfo/DumpTypeFromMangledName.swift]: Perhaps interleave the RUN commands with the CHECKs so they are grouped together?

File: test/DebugInfo/DumpTypeFromMangledName.swift [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+// CHECK-NEXT: (function_type escaping
+// CHECK-NEXT:   (input=tuple_type num_elements=0)
+// CHECK-NEXT:   (output=tuple_type num_elements=0))
+// CHECK-NEXT: Type of $SytD
+// CHECK-NEXT: (tuple_type num_elements=0)
+// CHECK-NEXT: Type of $Ss5Int32VD
+// CHECK-NEXT: (struct_type decl=Swift.(file).Int32)
+// CHECK-NEXT: Can't resolve type of $S4blah4mainyyF8PatatinoL_VMa
```
Comment (adrian-prantl) [inline test/DebugInfo/DumpTypeFromMangledName.swift]: I'm assuming this is a bug?

**Review by adrian-prantl — APPROVED** [review]

**Review by jrose-apple — COMMENTED** [review]
Looks good! Is it better to read in a file or just put the mangled name directly on the command line, though?

**Comment by dcci** [comment]
@jrose-apple I'm fine either way. I find that passing a list of names (maybe comma separated) would be doable but will make the test slightly harder to read (but I don't feel strong about it).

Originally I implemented this as an option which took a string rather than a file, but I figured it would be easier to batch so you don't have to pay the price of importing every time you want to look up a name.

@adrian-prantl I thought that as well, not sure whether type reconstruction isn't powerful enough or just isn't supposed to resolve this type (maybe @slavapestov knows) 

```
$ ./swift-demangle '$S4blah4mainyyF8PatatinoL_VMa'
$S4blah4mainyyF8PatatinoL_VMa ---> type metadata accessor for Patatino #1 in blah.main() -> ()
```

Please note all these `AT_linkage_name`(s) (including the one that's failing) are coming from the dwarfdump output of the test `.dSYM`, so, at least in theory, lldb could ask about them to type reconstruction.

**Comment by slavapestov** [comment]
@dcci I think the best approach is to have a file of input types, one per line, and an expected output file, and have a RUN: line that checks that the diff is empty. I'm not sure FileCheck is the best approach for this.

**Comment by dcci** [comment]
Cool, I'll modify this now. Thank you all for your comments

**Comment by adrian-prantl** [comment]
> I think the best approach is to have a file of input types, one per line, and an expected output file, and have a RUN: line that checks that the diff is empty. I'm not sure FileCheck is the best approach for this.
Wouldn't it be better to have input and expected output next to each other in the same file so it's easier to edit/read?

**Comment by jrose-apple** [comment]
I was thinking just `-lookup-type '$SSi' -lookup-type '$SSf'` or similar, but a file is fine. I just wanted to make sure you'd thought about both options and picked one.

**Comment by dcci** [comment]
@slavapestov updated to use the approach you suggested. Please let me know if this matches what you had in mind.
@adrian-prantl I moved the debug messages under `-verbose` otherwise the input/output file approach doesn't work. Hope this is fine with you.

**Comment by dcci** [comment]
@swift-ci please smoke test

**Comment by dcci** [comment]
@swift-ci please smoke test

**Pull request merged**